docs/adr/0034-rule-observer-callback.md

```
@@ -0,0 +1,100 @@
+
+> "I like the idea too but I was thinking we should move this to
+> experimental. We could
+> 1. Do not add the method in the interface but the type only
+> 2. create a function in
+> https://github.com/corazawaf/coraza/tree/main/experimental called
+> `WAFConfigWithRuleObserver` which does the interface assertion and calles
+> the type method.
+>
+> What do you think @heaven ?"
+> — @jcchavezs ([comment](https://github.com/corazawaf/coraza/pull/1478#issuecomment-3787116970))
+
```

markdownlint-cli2 (0.23.2)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)